runtime/compiler/control/rossa.cpp

```diff
@@ -1518,8 +1518,28 @@ onLoadInternal(
       {
       javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS;
       }
+#elif defined(J9HAMMER)
+   bool hasAVX = TR::Compiler->target.cpu.supportsAVX();
+   bool hasAVX512 = TR::Compiler->target.cpu.supportsFeature(OMR_FEATURE_X86_AVX512F);
+   bool hasAVX512bw = TR::Compiler->target.cpu.supportsFeature(OMR_FEATURE_X86_AVX512BW);
+   static char *useExtendedRegisters = feGetEnv("TR_UseExtendedVectorRegisters");
+
+   if (hasAVX512 && hasAVX512bw && useExtendedRegisters)
+      {
+      javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS;
+      javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_EXTENDED_VECTOR_REGISTERS;
+      }
+   else if (hasAVX512 && useExtendedRegisters)
+      {
+      javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_EXTENDED_VECTOR_REGISTERS;
+      }
+   else if (hasAVX)
+      {
+      javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS;
+      }
```

Comment:
I think to revert to the behaviour prior to the preservation PR the enablement logic should look like this.  Or am I wrong?  i.e., without the `TR_UseExtendedVectorRegisters` environment variable neither `J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS` nor `J9_EXTENDED_RUNTIME_USE_EXTENDED_VECTOR_REGISTERS` will be set.  In the preservation logic (one such example here [1]) if neither of these flags are set we will fall through to preserving just the 8 XMMs.

[1] https://github.com/eclipse-openj9/openj9/blob/a95344ad7bfafa579157b50583abcdb0f011b63e/runtime/oti/xhelpers.m4#L438

```
if (useExtendedRegisters)
   {
   if (hasAVX512 && hasAVX512bw)
      {
      ...
      }
   else if (hasAVX512)
      {
      ...
      }
   else if (hasAVX)
      {
      ...
      }
}
```

Reply:
I figured since there was no performance issue on AVX hardware, we could enable ymm0 - ymm15 by default

Reply:
I don't want to take that chance.  I'd like to disable it completely and then re-enable after we're sure it performs acceptably in all circumstances.